packages/teams-js/src/private/privateAPIs.ts

```diff
@@ -119,6 +119,24 @@ export function registerUserSettingsChangeHandler(
   registerHandler('userSettingsChange', handler, true, [settingTypes]);
 }
 
+/**
+ * @hidden
+ * register a handler to be called when a user's 3P cloud storage provider details changes. The changed setting type & value is provided in the callback.
+ *
+ * @param scenarioTypes - List of 3P cloud storage provider scenario changes to subscribe
+ * @param handler - When a subscribed setting is updated this handler is called
+ *
+ * @internal
+ */
+export function register3PCloudStorageProviderChangeHandler(
+  scenarioTypes: CloudStorageProvider3PScenario[],
+  handler: (scenarioType: CloudStorageProvider3PScenario, value: any) => void,
+): void {
+  ensureInitialized();
+
+  registerHandler('cloudStorageProvider3PChange', handler, true, [scenarioTypes]);
```

Comment:
Why do we need to introduce more complexity here with the scenarioTypes? I only see two scenario types and its unlikely that more will be added in the future. In either case, it would be better to have a separate handler registration method for each one, e.g.:
```
registerCloudStorageProvidersChanged
registerCloudStorageProviderContentChanged
```

Reply:
Was trying to avoiding creating two separate APIs for similar use case.
As per your suggestion, splitted the single method in two.